cli/src/commit_templater.rs

```diff
@@ -1381,6 +1437,70 @@ impl Template for Vec<Rc<CommitRef>> {
     }
 }
 
+/// Workspace name together with its working-copy commit for templating.
+#[derive(Debug, serde::Serialize)]
+pub struct WorkspaceRef {
+    /// Workspace name as a symbol.
+    name: RefSymbolBuf,
+    /// Working-copy commit of this workspace.
+    commit: Commit,
+}
+
+impl WorkspaceRef {
+    /// Creates a new workspace reference from the workspace name and commit.
+    pub fn new(name: WorkspaceNameBuf, commit: Commit) -> Rc<Self> {
+        Rc::new(WorkspaceRef {
+            name: RefSymbolBuf(name.into()),
+            commit,
+        })
+    }
+
+    /// Returns the workspace name symbol.
+    pub fn name(&self) -> &str {
+        self.name.as_ref()
+    }
+
+    /// Returns the working-copy commit of this workspace.
+    pub fn commit(&self) -> Commit {
+        self.commit.clone()
+    }
+}
+
+impl Template for Rc<WorkspaceRef> {
+    fn format(&self, formatter: &mut TemplateFormatter) -> io::Result<()> {
+        write!(formatter, "{}@", self.name)?;
+        Ok(())
+    }
+}
+
+impl Template for Vec<Rc<WorkspaceRef>> {
+    fn format(&self, formatter: &mut TemplateFormatter) -> io::Result<()> {
+        templater::format_joined(formatter, self, " ")
+    }
+}
+
+fn builtin_workspace_ref_methods<'repo>() -> CommitTemplateBuildMethodFnMap<'repo, Rc<WorkspaceRef>>
+{
+    let mut map = CommitTemplateBuildMethodFnMap::<Rc<WorkspaceRef>>::new();
+    map.insert(
+        "name",
+        |_language, _diagnostics, _build_ctx, self_property, function| {
+            function.expect_no_arguments()?;
+            let out_property = self_property.map(|ws_ref| ws_ref.name.clone());
```

Comment:
nit: and convert `name` to `RefSymbolBuf` buf here.

Reply:
I couldn't figure this out. I don't think I have a way of converting `&RefSymbol` to `RefSymbolBuf`, right?

I currently did, `RefSymbolBuf(ws_ref.name.as_str().into())` since doing `ws_ref.name.as_symbol()` was erroring out due to no abovementioned conversion.

Reply:
`RefSymbolBuf(ws_ref.name.into())`?